pinot-common/src/main/java/com/linkedin/pinot/common/utils/ServiceStatus.java

```diff
@@ -123,147 +128,130 @@ public String getStatusDescription() {
+        if (idealState == null || !idealState.isEnabled()) {
+          iterator.remove();
           continue;
         }
 
-        Map<String, String> statePartitionStateMap = getPartitionStateMap(helixState);
-        if(statePartitionStateMap.isEmpty() && !idealState.getPartitionSet().isEmpty()) {
-          _statusDescription = "statePartitionStateMapSize=" + statePartitionStateMap.size() + ", idealStateSize="
-              + idealState.getPartitionSet().size() + completedCountStr;
+        String descriptionSuffix =
+            String.format("(%s, %d/%d left)", resourceName, _resourcesToMonitor.size(), _numTotalResourcesToMonitor);
+        T helixState = getState(resourceName);
+        if (helixState == null) {
+          _statusDescription = "Helix state does not exist " + descriptionSuffix;
           return Status.STARTING;
         }
 
         // Check that all partitions that are supposed to be in any state other than OFFLINE have the same status in the
         // external view or went to ERROR state (which means that we tried to load the segments/resources but failed for
         // some reason)
-        for (String partition : idealState.getPartitionSet()) {
-          final String idealStateStatus = idealState.getInstanceStateMap(partition).get(_instanceName);
+        Map<String, String> partitionStateMap = getPartitionStateMap(helixState);
+        for (String partitionName : idealState.getPartitionSet()) {
+          String idealStateStatus = idealState.getInstanceStateMap(partitionName).get(_instanceName);
 
           // Skip this partition if it is not assigned to this instance or if the instance should be offline
           if (idealStateStatus == null || "OFFLINE".equals(idealStateStatus)) {
             continue;
           }
 
-          // If this instance state is not in the current state, then it hasn't finished starting up
-          if (!statePartitionStateMap.containsKey(partition)) {
-            _statusDescription = "statePartitionStateMap does not have " + partition + completedCountStr;
-            return Status.STARTING;
-          }
-
-          String currentStateStatus = statePartitionStateMap.get(partition);
-
           // If the instance state is not ERROR and is not the same as what's expected from the ideal state, then it
           // hasn't finished starting up
-          if (!"ERROR".equals(currentStateStatus) && !idealStateStatus.equals(currentStateStatus)) {
-            _statusDescription = partition +" currentStateStatus=" + currentStateStatus + ", idealStateStatus="
-                + idealStateStatus + completedCountStr;
-            return Status.STARTING;
+          String currentStateStatus = partitionStateMap.get(partitionName);
+          if (!idealStateStatus.equals(currentStateStatus)) {
+            if ("ERROR".equals(currentStateStatus)) {
+              LOGGER.error(String.format("Resource: %s, partition: %s is in ERROR state", resourceName, partitionName));
```

Comment:
This log message will be printed every time we call service status. Not fully convinced that we want that. Hopefully there are enough logs when the partition actually goes to ERROR state.